File: Day-01/q3-First and last occurrence of x/solution.java
public class day1_q3_solution {
    public static int first(int[] nums, int x){
        int ans=-1;
        int i=0;
        int j=nums.length-1;
        while(i<=j){
            int mid=(i+j)/2;
            if(nums[mid]>=x){
                ans=mid;
                j=mid-1;
            }
            else{
                i=mid+1;
            }
        }
        return ans;
    }
    public static int last(int[] nums, int x){
        int ans=-1;
        int i=0;
        int j=nums.length-1;
        while(i<=j){
            int mid=(i+j)/2;
            if(nums[mid]<=x){
                ans=mid;
                i=mid+1;
            }
            else{
                j=mid-1;
            }
        }
        return ans;
    }
    public static void main(String[] args) {
        int x=5;
        int[] nums={1, 3, 5, 5, 5, 5, 67, 123, 125};
        System.out.println(first(nums,x)+" "+last(nums,x));
    }
}

File: Day-01/q3-First and last occurrence of x/solution.cpp
vector<int> searchRange(vector<int>& nums, int target) {
        int startingPosition = -1, endingPosition = -1;
        vector<int> ans;
        int n = nums.size();
        for(int i=0; i<n; i++){
            if(nums[i] == target){
                startingPosition = i;
                break;
            }
        }
        for(int i=n-1; i>=0; i--){
            if(nums[i] == target){
                endingPosition = i;
                break;
            }
        }
        ans.push_back(startingPosition);
        ans.push_back(endingPosition);
        return ans;

}

// Another solution.cpp

class Solution {
public:
vector<int> searchRange(vector<int>& nums, int target) {        
vector<int> ans(2, -1);

        for(int i=0; i<nums.size(); i++){
            if(nums[i]==target){
                ans[0]=i;
                break;
            }
        }

        for(int i=0; i<nums.size(); i++){
            if(nums[i]==target)
            ans[1]=i;
        }

        return ans;
}
};

